.github/workflows/2_auto_publish_release.yml

```diff
@@ -0,0 +1,66 @@
+name: Release stage 2 - auto publish
+
+on:
+  pull_request:  # types AND paths
+    types: [closed]
+    paths: ['cylc/uiserver/__init__.py']
+    # NOTE: While this is too generic, we use the `if` condition of the job to narrow it down
+    # NOTE: Don't use `branches` as we might create release on any branch
+
+env:
+  # Best not to include the GH token here, only do it for the steps that need it
+  MERGE_SHA: ${{ github.event.pull_request.merge_commit_sha }}
+  CHANGELOG_FILE: CHANGES.md
+
+jobs:
+  publish:
+    if: >-  # NOTE: Can't use top-level env here unfortunately
+      github.event.pull_request.merged == true &&
+      contains(github.event.pull_request.labels.*.name, 'release')
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+
+    - name: Checkout repo
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ env.MERGE_SHA }}
+
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Get the version number
+      uses: cylc/release-actions/stage-2/get-version-from-pr@v1
+
+    - name: Build
+      uses: cylc/release-actions/build-python-package@v1
+
+    - name: Publish distribution to PyPI
+      uses: pypa/gh-action-pypi-publish@v1.4.2
+      with:
+        user: __token__ # uses the API token feature of PyPI - least permissions possible
+        password: ${{ secrets.PYPI_TOKEN }}
+
+    - name: Publish GitHub release
+      id: create-release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        commitish: ${{ env.MERGE_SHA }}
+        tag_name: ${{ env.VERSION }}
+        release_name: cylc-uiserver-${{ env.VERSION }}
+        prerelease: ${{ env.PRERELEASE }}
+        body: |
+          See [${{ env.CHANGELOG_FILE }}](https://github.com/${{ github.repository }}/blob/master/${{ env.CHANGELOG_FILE }}) for detail.
+        # TODO: Get topmost changelog section somehow and use that as the body?
```

Comment:
Also doesn't need to be addressed here, but here's my take (adapted from [this SO](https://stackoverflow.com/a/40450360)).

```
awk -F '[- ]' -v ver=0.4 '/^## / { if (p) { exit }; if ($4 == ver) { p=1; next } } p' CHANGES.md
```